pkg/designate/generate_bind9_pools_yaml.go

```diff
@@ -0,0 +1,215 @@
+	Targets     []Target
+	CatalogZone *CatalogZone // it is a pointer because it is optional
+}
+
+type NSRecord struct {
+	Hostname string
+	Priority int
+}
+
+type Nameserver struct {
+	Host string
+	Port int
+}
+
+type Target struct {
+	Type        string
+	Description string
+	Masters     []Master
+	Options     Options
+}
+
+type Master struct {
+	Host string
+	Port int
+}
+
+type Options struct {
+	Host           string
+	Port           int
+	RNDCHost       string
+	RNDCPort       int
+	RNDCConfigFile string
+}
+
+type CatalogZone struct {
+	FQDN    string
+	Refresh string
+}
+
+func GeneratePoolsYamlFile(pooslData []PoolData) (string, error) {
+	var buf bytes.Buffer
+	buf.WriteString("\n---\n")
+
+	for _, poolData := range pooslData {
+
+		// Create ns_records
+		nsRecords := []NSRecord{}
+		for hostname, priority := range poolData.NsRecordsMap {
+			nsRecord := NSRecord{
+				Hostname: hostname,
+				Priority: priority,
+			}
+			nsRecords = append(nsRecords, nsRecord)
+		}
+
+		// As far as I understand, we will only have one mdnsIP (primary DNS server)
+		var mdnsIP string
+		for _, v := range poolData.MdnsMap {
+			mdnsIP = v
+			break
+		}
+
+		// Create targets and nameservers
+		nameservers := []Nameserver{}
+		targets := []Target{}
+		rndcKeyNum := 1
+
+		for _, bindIP := range poolData.BindMap {
+			nameservers = append(nameservers, Nameserver{
+				Host: bindIP,
+				Port: 53,
+			})
+
+			target := Target{
+				Type:        "bind9",
+				Description: fmt.Sprintf("BIND9 Server %d (%s)", rndcKeyNum, bindIP),
+				Masters: []Master{
+					{
+						Host: mdnsIP,
+						Port: 16000, // TODO oschwart: change it to the real value
+					},
+					{
+						Host: mdnsIP,
+						Port: 16001, // TODO oschwart: change it to the real value
+					},
+					{
+						Host: mdnsIP,
+						Port: 16002, // TODO oschwart: change it to the real value
+					},
+				},
+				Options: Options{
+					Host:           bindIP,
+					Port:           53,
+					RNDCHost:       bindIP,
+					RNDCPort:       953,
+					RNDCConfigFile: fmt.Sprintf("%s/%s-%d.conf", RndcConfDir, DesignateRndcKey, rndcKeyNum),
+				},
+			}
+			targets = append(targets, target)
+			rndcKeyNum++
+		}
+
+		// Catalog zone is an optional section
+		// catalogZone := &CatalogZone{
+		// 	FQDN:    "example.org.",
+		// 	Refresh: "60",
+		// }
+
+		// Create the Pool struct with the dynamic values
+		pool := Pool{
+			Name:        "default",
+			Description: "DevStack BIND Pool",
+			Attributes:  poolData.Attributes,
+			NSRecords:   nsRecords,
+			Nameservers: nameservers,
+			Targets:     targets,
+			CatalogZone: nil, // set to catalogZone if this section should be presented
+		}
+
+		// Parse the template and return it as a string
+		tmpl, err := template.New("pool").Parse(poolsTemplate)
+		if err != nil {
+			return "", err
+		}
+
+		err = tmpl.Execute(&buf, pool)
+		if err != nil {
+			return "", err
+		}
+	}
+
+	return buf.String(), nil
+}
+
+const poolsTemplate = `
```

Comment:
this probably should not be in code but in a file in the templates directory. It's debatable whether it is any benefit outside of development (able to tweak template without recompiling) or checking a change at runtime but it will be consistent with other things that do this.

Reply:
Sounds good to me, I will move poolsTemplate to templates directory and import it from there.